### PR TITLE
Update terminology for #364

### PR DESCRIPTION
**Pull request by Gadgetoid**
Update problematic MISO/MOSI terminology as detailed here - https://www.oshwa.org/a-resolution-to-redefine-spi-signal-names

Pinout.xyz is a widely used educational resource, and thus should set the example.

This change is... significant. A lot of these terminology switches are the result of a find/replace operation, things might be overlooked. It will need testing before merging & pushing to live.

This may also have repurcussions on users ability to search for help/information when using outdated terminology and to recognise that `SDO==MOSI` and `SDI==MISO`. I don't know how significant this might be, but at least Pinout.xyz displays these new terms in a - familiar to some - context that may help.

### CODE REVIEW

**Comment by plugwash** [comment]
Note that SDO is not nessacerally the same as MOSI and SDI is not nessacerally.the same as MISO.

The big reason many designers preffer the MOSI/MISO terminology is it lets a signal have the same name as it flows through the system. With SDI/SDO the name changes at some point between the master and the slave and it can easily get confused over where.

File: src/es/overlay/uart.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ pin:
     direction: entrada
     active: alto (encendido)
 -->
-#UART - Universal Asynchronous Receiver/Transmitter (Receptor/Transmisor Universal Asíncrono)
+#UART - Universal Asynchronous Receiver/Transmitter (Receptor/TransSDIr Universal Asíncrono)
```
Comment (paulsnar) [inline src/es/overlay/uart.md:18]: Here's the other occurence of find-and-replace typos. At a quick glance seems like these are the only two though.

File: src/es/overlay/energenie-pi-mote.md [hunk 2]
```diff
@@ -44,6 +44,6 @@ pin:
 -->
 # Pi-mote
 
-Pi-Mote es una placa transmisora de RF que permite controlar hasta 4 zócalos Energenie de manera independiente utilizando Python.
+Pi-Mote es una placa transSDIra de RF que permite controlar hasta 4 zócalos Energenie de manera independiente utilizando Python.
```
Comment (paulsnar) [inline src/es/overlay/energenie-pi-mote.md:47]: Sorry about the previous mess of comments. Here's a place where the find-and-replace changed a word.

File: src/es/overlay/uart.md (hunk 1 above)
Comment (paulsnar) [inline src/es/overlay/uart.md:18]: And here's the other one.

```suggestion
#UART - Universal Asynchronous Receiver/Transmitter (Receptor/Transmisor Universal Asíncrono)
```

**Comment by lurch** [comment]
> This change is... significant.

Wow, indeed...! :eyes: 

> This may also have repurcussions on users ability to search for help/information when using outdated terminology and to recognise that SDO==MOSI and SDI==MISO.

Perhaps it's worth adding a note on the SPI page that "SDO was previously known as MOSI" or something? So that if users have a datasheet for a chip using the "old names" they _might_ be able to translate them to the "new names"? :shrug: :police_officer:

File: src/es/overlay/energenie-pi-mote.md (hunk 2 above)
Comment (Gadgetoid) [reply src/es/overlay/energenie-pi-mote.md:47]: Oh yeesh, thanks for picking up on this. That'll serve me right for doing case-insentitive search and replace.

**Comment by plugwash** [comment]
"SDO was previously known as MOSI" the problem is that statement is not correct in general.

SDO gives the direction relative to the current device, while MOSI reffers to the signal as a whole. So for a master SDO=MOSI and SDI=MISO but for a slave SDI=MOSI and SDO=MISO

**Comment by Gadgetoid** [comment]
@plugwash yeah, fully agree regarding the change from MOSI->MOSI to SDO->SDI. Moving to materially different terminology that doesn't work in quite the same way has the potential to be more damaging than the connotations of master/slave.

I think the potential repurcussions here are too complex to quantify, but that confusion is happening whether we change the terminology here or not.

Most of us have nearly a decade of experience dealing with learners in the Pi community- however- and we're intimately familiar with how terminology can trip up users, lead to false conclusions and- in the worst case- lead to a frustrated rage-quit from hobbyist tinkering. It's probably our duty to go the whole nine yards and *explain* this change properly on Pinout.xyz.

We can say "eh, it's not worth doing it on Pinout" but we can't stop this from happening elsewhere. Therefore the only rational course of action - to shelve the politics altogether - is to both make this change, and try to be a positive resource for explaining the differences in terminology and their implications.

We've got lovely protocol pages for that purpose, why not use 'em!

**Comment by lurch** [comment]
"**P**rimary **O**ut **P**eripherial **I**n" and "**P**eripheral **O**ut **P**rimary **I**n" ? :rofl: :rofl: 

P.S. How come this PR modifies `src/tr/tutorial/arduino-spi.md` but not the same page in any other languages?

**Comment by plugwash** [comment]
> We've got lovely protocol pages for that purpose, why not use 'em!
Agreed, whichever terminology is used on the main page it makes sense to have a protocol page that explains both sets of terminology. I highly doubt this campaign will cause the whole electronics world to transition.

**Comment by lurch** [comment]
I noticed last night that Wikipedia still uses the "old names" for [SPI](https://en.wikipedia.org/wiki/Serial_Peripheral_Interface), but also has a [separate page](https://en.wikipedia.org/wiki/Master/slave_(technology)) mentioning "Terminology concerns".

**Comment by Gadgetoid** [comment]
@lurch this PR missed a bunch of files that I've just discovered open in my IDE. Some of those accidentally replace the translated sections with English-language text. I need to step back and find the time to do this properly.

**Comment by lurch** [comment]
I just pointed a friend at this PR, and he pointed out that a simple search'n'replace isn't sufficient, because "MOSI" is now "SDO" from the perspective of the Pi, but "SDI" from the perspective of the HATs... :confused:

**Comment by Gadgetoid** [comment]
Let's all pretend "MOSI" and "MISO" stand for "Main-out Subordinate-in" and "Main-in Subordinate-out". Problem solved! 🤔

**Comment by Gadgetoid** [comment]
I'm inclined to close this and chalk it up as a bad job for the moment. It's difficult to balance the educational value of staying in the MOSI/MISO groove versus the potential exclusionary issues. I'm inclined to "lead the charge" as this is such a high-profile website in this community, but I still have doubts about the terminology change both technically and politically- this is a very Ameri-centric movement, and I am not an American, nor are an overwhelming majority of the people currently experiencing slavery.

Suffice to say I'm much more versed with the educational aspects and understand the repurcussions to beginner understanding that straying from terminology used elsewhere might have. I'm not wholly convinced that the benefits outweigh the drawbacks.

I remain open-minded and willing to be convinced otherwise.

**Pull request closed without merging**